File: src/shared/schemas/training.ts
import { z } from 'zod'

const algorithms = ['random_forest', 'svm', 'logistic_regression', 'xgboost'] as const

export const trainModelSchema = z.object({
  dataset_id: z.string().min(1, 'Dataset is required'),
  algorithm: z.enum(algorithms, { message: 'Invalid algorithm' }),
  pipeline_id: z.string().optional(),
  target_column: z.string().optional(),
  test_size: z.number().min(0).max(1).optional().default(0.2),
  random_seed: z.number().int().optional().default(42),
  hyperparameters: z.record(z.unknown()).optional(),
  name: z.string().max(255).optional(),
})

export type TrainModelFormData = z.infer<typeof trainModelSchema>

File: backend/app/api/v1/schemas/training.py
from pydantic import BaseModel, Field


class TrainModelSchema(BaseModel):
    dataset_id: str = Field(min_length=1)
    algorithm: str = Field(pattern=r"^(random_forest|svm|logistic_regression|xgboost)$")
    pipeline_id: str | None = None
    target_column: str | None = None
    test_size: float = Field(default=0.2, ge=0, le=1)
    random_seed: int = Field(default=42, ge=0)
    hyperparameters: dict = Field(default_factory=dict)
    name: str | None = None
